ilc/client/handlePageTransaction.spec.js

```diff
@@ -0,0 +1,278 @@
+                    <div id="${slots.body.id}"></div>
+                </div>
+            `;
+        },
+        navbar: {
+            id: 'navbar',
+            getComputedStyle: () => window.getComputedStyle(slots.navbar.ref, null),
+        },
+        body: {
+            id: 'body',
+            getComputedStyle: () => window.getComputedStyle(slots.body.ref, null),
+            getAttributeName: () => document.body.getAttribute('name'),
+        },
+    };
+
+    const applications = {
+        navbar: {
+            id: 'navbar-application',
+            class: 'navbar-spa',
+            appendApplication: () => slots.navbar.ref.appendChild(applications.navbar.ref),
+            resetRef: () => {
+                applications.navbar.ref = html`
+                    <div id="${applications.navbar.id}" class="${applications.navbar.class}">
+                        Hello! I am Navbar SPA
+                    </div>
+                `;
+            },
+        },
+        body: {
+            id: 'body-application',
+            class: 'body-spa',
+            appendApplication: () => slots.body.ref.appendChild(applications.body.ref),
+            removeApplication: () => slots.body.ref.removeChild(applications.body.ref),
+            resetRef: () => {
+                applications.body.ref = html`
+                    <div id="${applications.body.id}" class="${applications.body.class}">
+                        Hello! I am Body SPA
+                    </div>
+                `;
+            },
+        },
+    };
+
+    const spinner = {
+        id: 'ilc-spinner',
+        class: 'ilc-spinner',
+        getRef: () => document.getElementById(spinner.id),
+    };
+
+    let clock;
+
+    beforeEach(() => {
+        window.location.hash = locationHash;
+        window.ilcConfig = {
+            tmplSpinner: `<div id="${spinner.id}" class="${spinner.class}">Hello! I am Spinner</div>`,
+        };
+
+        slots.resetRef();
+        applications.body.resetRef();
+        applications.navbar.resetRef();
+
+        slots.appendSlots();
+
+        slots.navbar.ref = document.getElementById(slots.navbar.id);
+        slots.body.ref = document.getElementById(slots.body.id);
+
+        clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+        slots.removeSlots();
+        clock.restore();
+    });
+
+    it('should throw an error when a slot name is not provided', async () => {
+        let caughtError;
+
+        try {
+            handlePageTransaction();
+        } catch(error) {
+            caughtError = error;
+        }
+
+        chai.expect(caughtError).to.be.an.instanceof(Error);
+        chai.expect(caughtError.message).to.be.eql('A slot name was not provided!');
+    });
+
+    it('should throw an error when a slot action does not match any possible option to handle', async () => {
+        let caughtError;
+
+        try {
+            handlePageTransaction(slots.body.id, 'undefined');
```

Comment:
`chai.expect(() => handlePageTransaction(slots.body.id, 'undefined')).to.throw()`
https://www.chaijs.com/api/bdd/#method_throw
Instead of try/catch